### PR TITLE
Override in the framework/src/base directory

### PR DESCRIPTION
**Pull request by permcody**
Part 2 of 2 for the override keyword. This adds the remaining overrides to the framework folder in the base directory. I was much more conservative on this set of changes (only deleted the MooseInit() destructor.

Let's get this merged before we start tiger team!
@idaholab/moose-team 

refs #7171

### CODE REVIEW

File: framework/include/base/CacheChangedListsThread.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ class CacheChangedListsThread : public ThreadedElementLoopBase<ConstElemRange>
   CacheChangedListsThread(CacheChangedListsThread & x, Threads::split split);
   virtual ~CacheChangedListsThread();
 
-  virtual void onElement(const Elem *elem);
+  virtual void onElement(const Elem *elem) override;
```
Comment (jwpeterson) [inline framework/include/base/CacheChangedListsThread.h]: Might as well fix pointer declaration spacing if you are changing this line.

File: framework/include/base/ComputeDiracThread.h [hunk 2]
```diff
@@ -39,11 +39,11 @@ class ComputeDiracThread : public ThreadedElementLoop<DistElemRange>
...
   virtual ~ComputeDiracThread();
 
-  virtual void subdomainChanged();
-  virtual void pre();
-  virtual void onElement(const Elem *elem);
-  virtual void postElement(const Elem * /*elem*/);
-  virtual void post();
+  virtual void subdomainChanged() override;
+  virtual void pre() override;
+  virtual void onElement(const Elem *elem) override;
```
Comment (jwpeterson) [inline framework/include/base/ComputeDiracThread.h]: `*` spacing

File: framework/include/base/ComputeElemAuxVarsThread.h [hunk 3]
```diff
@@ -36,9 +36,9 @@ class ComputeElemAuxVarsThread : public ThreadedElementLoop<ConstElemRange>
 
   virtual ~ComputeElemAuxVarsThread();
 
-  virtual void subdomainChanged();
-  virtual void onElement(const Elem *elem);
-  virtual void post();
+  virtual void subdomainChanged() override;
+  virtual void onElement(const Elem *elem) override;
```
Comment (jwpeterson) [inline framework/include/base/ComputeElemAuxVarsThread.h]: `*` spacing

File: framework/include/base/ComputeElemDampingThread.h [hunk 4]
```diff
@@ -36,7 +36,7 @@ class ComputeElemDampingThread : public ThreadedElementLoop<ConstElemRange>
 
   virtual ~ComputeElemDampingThread();
 
-  virtual void onElement(const Elem *elem);
+  virtual void onElement(const Elem *elem) override;
```
Comment (jwpeterson) [inline framework/include/base/ComputeElemDampingThread.h]: spacing

File: framework/include/base/ComputeIndicatorThread.h [hunk 5]
```diff
@@ -40,12 +40,12 @@ class ComputeIndicatorThread : public ThreadedElementLoop<ConstElemRange>
...
   virtual ~ComputeIndicatorThread();
 
-  virtual void subdomainChanged();
-  virtual void onElement(const Elem *elem);
-  virtual void onBoundary(const Elem *elem, unsigned int side, BoundaryID bnd_id);
-  virtual void onInternalSide(const Elem *elem, unsigned int side);
-  virtual void postElement(const Elem * /*elem*/);
-  virtual void post();
+  virtual void subdomainChanged() override;
+  virtual void onElement(const Elem *elem) override;
```
Comment (jwpeterson) [inline framework/include/base/ComputeIndicatorThread.h]: spacing these 3 lines

File: framework/include/base/ComputeJacobianThread.h [hunk 6]
```diff
@@ -38,13 +38,13 @@ class ComputeJacobianThread : public ThreadedElementLoop<ConstElemRange>
...
-  virtual void subdomainChanged();
-  virtual void onElement(const Elem *elem);
-  virtual void onBoundary(const Elem *elem, unsigned int side, BoundaryID bnd_id);
-  virtual void onInternalSide(const Elem *elem, unsigned int side);
-  virtual void onInterface(const Elem *elem, unsigned int side, BoundaryID bnd_id);
-  virtual void postElement(const Elem * /*elem*/);
-  virtual void post();
+  virtual void subdomainChanged() override;
+  virtual void onElement(const Elem *elem) override;
+  virtual void onBoundary(const Elem *elem, unsigned int side, BoundaryID bnd_id) override;
```
Comment (jwpeterson) [inline framework/include/base/ComputeJacobianThread.h]: spacing on 3 lines

**Comment by jwpeterson** [comment]
Basically text replace "Elem *elem" with "Elem * elem" and I think you're good.

**Comment by permcody** [comment]
Yeah, I looked at the whole patch. Looks like only `* elem` issues.

**Pull request merged**